 hive/release-0.9.0/bin/hive | 44 +++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 43 insertions(+), 1 deletion(-)

diff --git a/hive/release-0.9.0/bin/hive b/hive/release-0.9.0/bin/hive
index edac15c..49fd19d 100644
--- a/hive/release-0.9.0/bin/hive
+++ b/hive/release-0.9.0/bin/hive
@@ -20,7 +20,37 @@ case "`uname`" in
    CYGWIN*) cygwin=true;;
 esac
 
-bin=`dirname "$0"`
+BASEMAPR=${MAPR_HOME:-/opt/mapr}
+export HADOOP_HOME=${BASEMAPR}/hadoop/hadoop-0.20.2
+MAPR_HBASE_HOME=${MAPR_HBASE_HOME:-${BASEMAPR}/hbase/hbase-0.92.0/}
+MAPR_ZK_HOME=${MAPR_ZK_HOME:-${BASEMAPR}/lib}
+
+function get_canonical_dir() {
+  target="$1"
+  canonical_name=`readlink -f ${target} 2>/dev/null`
+  if [[ $? -eq 0 ]]; then
+    canonical_dir=`dirname $canonical_name`
+    echo ${canonical_dir}
+    return
+  fi
+  
+  # Mac has no readlink -f
+  cd `dirname ${target}`
+  target=`basename ${target}` 
+  
+  # chase down the symlinks
+  while [ -L ${target} ]; do
+    target=`readlink ${target}`
+    cd `dirname ${target}`
+    target=`basename ${target}`
+  done
+
+  canonical_dir=`pwd -P`
+  ret=${canonical_dir}
+  echo $ret
+}
+
+bin=$(get_canonical_dir "$0")
 bin=`cd "$bin"; pwd`
 
 . "$bin"/hive-config.sh
@@ -94,6 +124,18 @@ for f in ${HIVE_LIB}/*.jar; do
   CLASSPATH=${CLASSPATH}:$f;
 done
 
+# Add mapr hbase, zookeeper path overrides any jars in hive/lib
+if [ -f "${MAPR_HBASE_HOME}/hbase-0.92.0.jar" ] ; then
+  CLASSPATH=${CLASSPATH}:${MAPR_HBASE_HOME}/hbase-0.92.0.jar;
+fi
+
+if [ -f "${MAPR_HBASE_HOME}/hbase-0.92.0-tests.jar" ] ; then
+  CLASSPATH=${CLASSPATH}:${MAPR_HBASE_HOME}/hbase-0.92.0-tests.jar;
+fi
+
+if [ -f "${MAPR_ZK_HOME}/zookeeper-3.3.2.jar" ] ; then
+  CLASSPATH=${CLASSPATH}:${MAPR_ZK_HOME}/zookeeper-3.3.2.jar;
+fi
 # add the auxillary jars such as serdes
 if [ -d "${HIVE_AUX_JARS_PATH}" ]; then
   for f in ${HIVE_AUX_JARS_PATH}/*.jar; do
